### ISSUE TITLE
Patch Kexts Wont Install After Upgrading to Big Sur 11.4 Error 1x1

### ISSUE THREAD

**Issue by michaiel** [BenSova/Patched-Sur#609]
I upgraded to Big Sur on my 2012 MacBook Pro and everything worked until I attempted to install "Patch Kexts" from Patched Sur on the USB drive.

Patched Sur did NOT install to the Applications folder when I updated to 11.4 from Catalina.
I have attempted to copy Patched Sur to Applications, but still doesn't work.
I downloaded Patched Sur and attempted to install "Patched Kexts", but still get the same error.

The red text says "You can fix this by booting into the purple EFI Boot on your Patched Installer USB.
When I attempt to do this the screen just goes black and I have to restart the computer again.

Please advise...

thanks

**Comment by Monkiey** [comment]
Yes the computer is supposed to do that by shutting down after you do the EFI, and then you restarted which is correct you restarted, now go to the application and try to patch kexts again.

**Comment by michaiel** [comment]
I tried restarting several times, but I keep getting the same 1x1 error...
The one thing I noticed is that Patched Sur did not install into the Applications folder so I have been using the one on the usb flash drive.
I downloaded another copy of Patched Sur, but it did not work either...

**Comment by ozziepadilla** [comment]
Go here and download the post install app https://github.com/BenSova/Patched-Sur

**Comment by michaiel** [comment]
I have Patched Sur 0.2.1

**Comment by michaiel** [comment]
Unfortunately I get the same error after downloading again and attempting to install

**Comment by michaiel** [comment]
The red text says "You can fix this by booting into the purple EFI Boot on your Patched Installer USB.
When I attempt to do this the screen just goes black and I have to restart the computer again.

I just cant get it to boot into the EFI

**Comment by Monkiey** [comment]
Efi boot does not boot into anything it just set commands for your mac

**Comment by Monkiey** [comment]
see here from the readme: 
"11. Select the purple EFI Boot drive, (if there are multiple unplug and replug in your drive and select the EFI Boot that disappeared and reappeared). Then, your Mac will quickly turn off, so turn it back on while, again, holding down Option/Alt. Then, select Install macOS Big Sur."

**Comment by michaiel** [comment]
so I need to reinstall Big Sur again?

**Comment by Monkiey** [comment]
> so I need to reinstall Big Sur again?

No, all you have too is Revoot hold option clicking EFI boot than restart your mac and got to patch kexts in the post-install and try again.

**Comment by michaiel** [comment]
ok...thanks for clarifying...will do and report in to everyone...thanks

**Comment by michaiel** [comment]
Unfortunately it still wont install the kexts correctly...1x1

**Comment by Monkiey** [comment]
You can try to reinstall macOS?

**Comment by BenSova** [comment]
Did you create that USB on a late 2013 iMac?

**Comment by BenSova** [comment]
I'd wait before reinstalling macOS

**Comment by michaiel** [comment]
I did create the usb on a late 2013 iMac...uh oh?

**Comment by Monkiey** [comment]
> I did create the usb on a late 2013 iMac...uh oh?

You need to create the installer from the mac you are using the 2012 mbp

**Comment by michaiel** [comment]
So I need to create separate installers for each computer?
I have another MacBook Pro here I can do that and report in.

Thanks

**Comment by michaiel** [comment]
one question...can I reinstall over the existing Big Sur when I do this?
Or do I have to back up on the OS and start over?

**Comment by Monkiey** [comment]
> So I need to create separate installers for each computer?
> I have another MacBook Pro here I can do that and report in.
> 
> Thanks

You could also  just do it on the same machine that you are trying to patch

**Comment by Monkiey** [comment]
> one question...can I reinstall over the existing Big Sur when I do this?
> Or do I have to back up on the OS and start over?

Just install it over existing, should work

**Comment by michaiel** [comment]
I dont have internet on the mb yet...need the kexts that wont install

**Comment by Monkiey** [comment]
> I dont have internet on the mb yet...need the kexts that wont install

Yep your are right totally forgot lol

**Comment by BenSova** [comment]
> I did create the usb on a late 2013 iMac...uh oh?

The EFI Boot works differently when made on those Macs, hence why it's not fixing it. No worries though, we can still fix your problem. 

1. Boot into Internet Recovery Mode (CMD-OPTION-R on boot)
2. Open Terminal from the Utilities menu
3. Type in `nvram csr-active-config=%7f%08%00%00` then press enter
4. Reboot your Mac, and try patching kexts again.

**Comment by michaiel** [comment]
Ben, thank you for reply! will report in and let everyone know...

**Comment by michaiel** [comment]
I get ...

nvram: Error setting variable - 'csr-active-config': (iokit/common) not permitted
-bash-3.2#

**Comment by BenSova** [comment]
Try that again, but use CMD-OPTION-SHIFT-R instead, that should give you Mountain Lion recovery which was way before SIP was implemented and which should prevent any permission denied errors.

**Comment by michaiel** [comment]
I get ...

csr-active-config=%7f%08%00%00: command not found
-bash-3.2#

**Comment by Monkiey** [comment]
Try to patch big sur recovery mode by going into the post-install and clicking the option install-recovery.

**Comment by michaiel** [comment]
Install Recovery says the SIP is on...do you know where in recovery mode I could do it?

**Comment by Monkiey** [comment]
> Install Recovery says the SIP is on...do you know where in recovery mode I could do it?

You can try to go back to the previous recoveries and try to disable sip, however, it might give you a similar error to the car config.

**Comment by BenSova** [comment]
> I get ...
> 
> 
> 
> csr-active-config=%7f%08%00%00: command not found
> 
> -bash-3.2#

You typed `csr-active-config=%7f%08%00%00` not `nvram csr-active-config=%7f%08%00%00`

**Comment by michaiel** [comment]
sorry I typed it wrong...I got patch recovery to finish patching...

**Comment by michaiel** [comment]
ok...ran Patch Sur again and the Kexts worked!
Restarted computer and I have internet again...
Thanks so much for this...Big shout out to BenSova and Monkiey!

**Comment by Monkiey** [comment]
> ok...ran Patch Sur again and the Kexts worked!
> Restarted computer and I have internet again...
> Thanks so much for this...Big shout out to BenSova and Monkiey!

Glad it worked, no problem
